app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb

```diff
@@ -212,14 +214,22 @@ def allowed_hosts_obj(_options = {})
 
   def allowed_storages(options = {})
     return [] if (src = resources_for_ui).blank?
-    result = super
+    storages = super
+    storages = storages.select { |s| s.storage_domain_type == "data" }
+    src_template = load_ar_obj(src[:vm])
+    if selected_linked_clone?
+      s_id = src_template.storage_id
+      return [storages.detect { |s| s.id == s_id }]
```

Comment:
Can you `[].compact` this in case the storage detect doesn't find the storage for some reason and returns nil

Reply:
Or continue to use `.select` which will return `[]` if nothing is found even though it is weird to use select to find by id

Reply:
right! done.